src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -439,24 +439,25 @@ void insert_reorders_in_dir(program& p, const std::map<program_node*, format::ty
         auto in_layout = predecessor->get_output_layout();
         auto out_layout = in_layout;
         in_layout.format = get_target_output_format(lo, fmt_map, predecessor);
-        auto target_input0_format = get_target_input0_format(lo, fmt_map, successor);
+        auto target_input_format = get_target_input_format(lo, fmt_map, successor);
 
         for (auto& fused_prim : successor->get_fused_primitives()) {
             // If it is input of fused node, use output layout instead of input layout
             if (successor->get_dependencies().size() <= fused_prim.dep_start_idx)
-                continue;
+                break;
+
             auto& dependency = successor->get_dependency(fused_prim.dep_start_idx);
             if (&dependency == predecessor) {
-                target_input0_format = get_target_output_format(lo, fmt_map, successor);
+                target_input_format = get_target_output_format(lo, fmt_map, successor);
                 GPU_DEBUG_IF(debug_config->verbose >= 2) {
                     GPU_DEBUG_COUT << __func__ << ":" << __LINE__ << ": Use output format of successor " << successor->id() << " : "
-                                << fmt_to_str(target_input0_format) << std::endl;
+                                << fmt_to_str(target_input_format) << std::endl;
                 }
                 break;
             }
         }
```

Comment:
With multiple input support, I think now we can simplify this code like this:

```
auto input_idx = predecessor->get_input_index(successor)
target_input_format = get_target_input_format(lo, fmt_map, successor, input_idx)
```

We need to set proper input index in `select_preferred_formats`. That is, for fused input ports, set input layout same as output layout.